.github/workflows/dev-docs-sync.yml

```diff
@@ -0,0 +1,62 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+name: Dev docs sync
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 1 * * *" # Every day at 1 am UTC
```

Comment:
Our other jobs run at midnight, so I figured this reduces contention such as if we get rate limited by GitHub by our external link checker